Vue 1220 header create shared placeholder component

dyersituations opened the pull request:
https://kiva.atlassian.net/browse/VUE-1220

- Basic placeholder ported from `ui` repo

![image](https://user-images.githubusercontent.com/16867161/184024704-1bc9bf0d-951f-42a0-801a-fdc5f67456d6.png)

mcstover commented:
@dyersituations I know I've approved already but should/could we add another story that helps cover the dynamic style, dynamic class and static class attributes that live on the original version? Seems like those attributes are needed now but just want to see if it's working ok...

dyersituations commented:
> @dyersituations I know I've approved already but should/could we add another story that helps cover the dynamic style, dynamic class and static class attributes that live on the original version? Seems like those attributes are needed now but just want to see if it's working ok...

When I ported this component over, I changed it from a `functional` component, since those aren't supported by Vue3. This is a simple enough component that I figured there shouldn't be a huge performance difference. I updated the story to show that both static and non-static versions of `class` and `style` work.

mcstover reviewed (approved):
Cool thanks!
